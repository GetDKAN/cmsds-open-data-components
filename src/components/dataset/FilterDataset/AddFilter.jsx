import React from 'react'
import PropTypes from 'prop-types'
import FontAwesomePro from '../../common/FontAwesomePro/FontAwesomePro'

const AddFilter = ({ addFilter }) => {
  return (
    <button className='add-filter' aria-label='Add filter' onClick={addFilter}>
      <FontAwesomePro icon='plus-circle' size={16} />
      <span>Add new filter</span>
    </button>
  )
}

AddFilter.propTypes = {
  /**
   * Callback function to set filter
   */
  addFilter: PropTypes.func
}

AddFilter.displayName = 'AddFilter'
export default AddFilter
