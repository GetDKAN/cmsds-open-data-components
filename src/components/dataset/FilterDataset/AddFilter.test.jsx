import { render, fireEvent, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import AddFilter from './AddFilter'

describe('AddFilter component.', () => {
  it('Matches snapshot.', () => {
    const renderedAddFilter = renderer.create(
      <AddFilter addFilter={() => {}} />
    ).toJSON()

    expect(renderedAddFilter).toMatchSnapshot()
  })

  it('\'addFilter\' prop is called when button is clicked.', () => {
    const addFilter = vi.fn()
    
    render(
      <AddFilter addFilter={addFilter} />
    )

    fireEvent.click(screen.getByRole('button', { name: 'Add filter' }))

    expect(addFilter).toHaveBeenCalled()
  })
})
