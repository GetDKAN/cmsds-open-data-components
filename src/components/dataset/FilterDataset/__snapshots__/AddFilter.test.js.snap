// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFilter component. Matches snapshot. 1`] = `
<button
  aria-label="Add filter"
  className="add-filter"
  onClick={[Function]}
>
  <svg
    aria-label="plus circle"
    className="fa fa-plus-circle"
    data-icon="plus-circle"
    data-prefix="fal"
    focusable="false"
    height={16}
    role="img"
    viewBox="0 0 512 512"
    width={16}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      className=""
      d="M384 250v12c0 6.6-5.4 12-12 12h-98v98c0 6.6-5.4 12-12 12h-12c-6.6 0-12-5.4-12-12v-98h-98c-6.6 0-12-5.4-12-12v-12c0-6.6 5.4-12 12-12h98v-98c0-6.6 5.4-12 12-12h12c6.6 0 12 5.4 12 12v98h98c6.6 0 12 5.4 12 12zm120 6c0 137-111 248-248 248S8 393 8 256 119 8 256 8s248 111 248 248zm-32 0c0-119.9-97.3-216-216-216-119.9 0-216 97.3-216 216 0 119.9 97.3 216 216 216 119.9 0 216-97.3 216-216z"
      fill="currentColor"
    />
  </svg>
  <span>
    Add new filter
  </span>
</button>
`;
